File: 1624-largest-substring-between-two-equal-characters/1624-largest-substring-between-two-equal-characters.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        
        HashMap< Character, Integer> m=new HashMap<>();
        int ans=0 , n=s.length();
        
        for( int i=0;i<n;i++){
            
            char l=s.charAt(i);
            char r=s.charAt(n-i-1);
        
            if( m.containsKey(l)){
                
                ans=Math.max( ans , Math.abs(m.get(l)-i));
                
                m.put(l, Math.min(i, m.get(l)));
            }
            else{
                m.put(l,i);
            }
            
            if(m.containsKey(r)){
                ans=Math.max( ans , Math.abs(m.get(r)-(n-i-1)));
                
                 m.put(r, Math.max(n-i-1, m.get(r)));
            }
            else{
                m.put(r,(n-i-1));
            }
        }
        return ans-1;
        
    }
}

File: 1624-largest-substring-between-two-equal-characters/1624-largest-substring-between-two-equal-characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        
        unordered_map< char , int> m;
        int f[26]={0};
        int ans=0;
        
        for( int i=0; i<s.size() ;i++){
            if(m.count(s[i]))
            m[s[i]]=min( m[s[i]], i);
            else m[s[i]]=i;
        }
        for( int i=s.size()-1;i>=0;i--){
            ans=max( ans , i-m[s[i]]);
        }
        return ans-1;
    }
};